File: app/src/main/java/com/brunel/group30/fitnessapp/CustomNumberPicker.java
package com.brunel.group30.fitnessapp;

import android.content.Context;
import android.content.res.TypedArray;
import android.util.AttributeSet;
import android.widget.NumberPicker;

public class CustomNumberPicker extends NumberPicker {
    public CustomNumberPicker(Context context, AttributeSet attrs) {
        super(context, attrs);
        processAttributeSet(attrs);
    }

    private void processAttributeSet(AttributeSet attrs) {
        TypedArray attributes = getContext().getTheme().obtainStyledAttributes(attrs,
                R.styleable.CustomNumberPicker, 0, 0);

        try {
            this.setMinValue(attributes.getInt(R.styleable.CustomNumberPicker_minValue, 0));
            this.setMaxValue(attributes.getInt(R.styleable.CustomNumberPicker_maxValue, 0));
            this.setValue(attributes.getInt(R.styleable.CustomNumberPicker_defaultValue,
                    attributes.getInt(R.styleable.CustomNumberPicker_minValue, 0)));
        } finally {
            attributes.recycle();
        }
    }
}

File: app/src/main/java/com/brunel/group30/fitnessapp/Custom/CustomNumberPicker.kt
package com.brunel.group30.fitnessapp.Custom

import android.content.Context
import android.util.AttributeSet
import android.widget.NumberPicker
import com.brunel.group30.fitnessapp.R
import java.text.NumberFormat
import java.util.*


class CustomNumberPicker(context: Context, attrs: AttributeSet) : NumberPicker(context, attrs) {
    var originalMinValue = 0
    var originalMaxValue = 0

    init {
        processAttributeSet(attrs)
    }

    private fun processAttributeSet(attrs: AttributeSet) {
        val attributes = context.theme.obtainStyledAttributes(attrs,
                com.brunel.group30.fitnessapp.R.styleable.CustomNumberPicker, 0, 0)

        try {
            this.minValue = attributes.getInt(R.styleable.CustomNumberPicker_minValue, 0)
            this.maxValue = attributes.getInt(R.styleable.CustomNumberPicker_maxValue, 0)
            this.value = attributes.getInt(R.styleable.CustomNumberPicker_defaultValue,
                    attributes.getInt(R.styleable.CustomNumberPicker_minValue, 0))

            this.originalMinValue = this.minValue
            this.originalMaxValue = this.maxValue
        } finally {
            attributes.recycle()
        }
    }

    fun getArrayWithSteps(iStep: Int, unit: String?): Array<String?> {
        val iStepsArray = (this.maxValue - this.minValue) / iStep + 1

        val arrayValues = arrayOfNulls<String>(iStepsArray)
        for (i in 0 until iStepsArray) {
            arrayValues[i] = NumberFormat.getNumberInstance(Locale.getDefault())
                    .format(this.minValue + i * iStep).toString() + " " + unit
        }

        this.minValue = 0
        this.maxValue = arrayValues.size - 1

        return arrayValues
    }
}
